woocommerce/woocommerce
# [Customize Your Store] Loading - Persisting AI assembled site

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes #40163.

Update block theme global styles and block template data with AI response dat



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

This feature is part of the larger customize your store development and behind a feature flag, hence QA review can be deferred until the call for testing before release.

Make sure you are testing this with a Jetpack connected instance of WooCommerce, as the AI text completion endpoint can only be called with a Jetpack connected account.

1. Create a new WooCommerce installation with this version.
2. Make sure to enable `customize-store` feature flag
3. Go to `WooCommerce -> Home` and click `Customize Store` task
4. Click `Design with AI` button
5. Follow through the flow all the way till the loader shows up
6. Should be redirected to assembler hub screen
7. Examine that the theme style and template are set based on AI response.
  
This can be inspected using the XState inspector:

![Screenshot 2023-09-20 at 13 37 39](https://github.com/woocommerce/woocommerce/assets/4344253/ea4fc87a-7c90-4157-96aa-1273800feda8)

https://github.com/woocommerce/woocommerce/assets/4344253/a60b1b98-5e54-4c09-bead-6297a455010c




### Changelog entry




<details>

#### Significance




#### Type




#### Message 

Persist CYS AI assembled site

#### Comment 

</details>
